### PR TITLE
upgrade 2-advanced/dubbo-samples-notify to jdk17 and spring3

### PR DESCRIPTION
**Pull request by hhuang1231**
- upgrade jdk17 and springboot3
- upgrade to junit5
- Clean up commented out code

### CODE REVIEW

File: 2-advanced/dubbo-samples-notify/dubbo-samples-notify-consumer/src/test/java/org/apache/dubbo/samples/notify/consumer/NotifyIT.java [hunk 1]
```diff
@@ -17,31 +17,27 @@
...
 @SpringBootTest
-@RunWith(SpringRunner.class)
 public class NotifyIT {
-    // @DubboReference
+
     @DubboReference(timeout = 6000, methods = @Method(name = "sayHello", onreturn = "notify.onReturn", onthrow = "notify.onThrow"))
     private DemoService demoService;
 
-    @Autowired
+    @Resource
```
Comment (yuluo-yx) [inline 2-advanced/dubbo-samples-notify/dubbo-samples-notify-consumer/src/test/java/org/apache/dubbo/samples/notify/consumer/NotifyIT.java:34]: Why did you change to use the resource annotation?

**Review by yuluo-yx — COMMENTED** [review]
We should make PR changes as homogeneous as possible, with one PR accomplishing only one thing if there are no other associated factors.

**Comment by chickenlj** [comment]
Please check the comment here and update dependency in pom.xml

**Comment by hhuang1231** [comment]
> Please check the comment here and update dependency in pom.xml [#1055 (comment)](https://github.com/apache/dubbo-samples/pull/1055#issuecomment-1996426918)

@chickenlj Now I have modified the module to follow the format of POM in #1052

**Review by chickenlj — APPROVED** [review]
LGTM.
